streams-tests/jvm/src/test/scala/zio/stream/StreamPullSafetySpec.scala

```diff
@@ -63,6 +63,29 @@ object StreamPullSafetySpec extends ZIOBaseSpec {
         } yield assert(fin, isTrue) && assert(pulls, equalTo(List(Left(Some("Ouch")), Left(None), Left(None))))
       }
     ),
+    suite("Stream.flatten")(
+      testM("is safe to pull again after success") {
+        Stream
+          .flatten(Stream(Stream.fromEffect(UIO.succeed(5))))
+          .process
+          .use(nPulls(_, 3))
+          .map(assert(_, equalTo(List(Right(5), Left(None), Left(None)))))
```

Comment:
not that it matters but you can write as:
```scala
assertM(
  Stream.flatten(Stream(Stream.fromEffect(UIO.succeed(5)))).process.use(nPulls(_, 3)),
  equalTo(List(Right(5), Left(None), Left(None))
)
```

Reply:
I'm aware, however your formulation tends to show the warning "type inferred to be Any" for the environment, which has since been fixed in upstream scalac. I mainly avoid it for the warning.